app/code/Magento/CatalogSearch/Model/ResourceModel/Fulltext/Collection.php

```diff
@@ -344,8 +344,6 @@ protected function _renderFiltersBefore()
             []
         );
 
-        $this->_totalRecords = $this->searchResult->getTotalCount();
-
```

Comment:
> Fulltext results count can be more than original select results count (joinInner is used). Leave $this->_totalRecords empty to initialize later, after actual data have been loaded.

potentially it can. But in fact join
```
        $table = $temporaryStorage->storeApiDocuments($this->searchResult->getItems());

        $this->getSelect()->joinInner(
            [
                'search_result' => $table->getName(),
            ],
            'e.entity_id = search_result.' . TemporaryStorage::FIELD_ENTITY_ID,
            []
        );

```
is just a performance optimization for query
`select * from catalog_product_flat where entity_id IN (list_of_10000_ids)`
because inner join for temporary table which contains just IDs column with Primary Key on it is much faster than IN clause.

If we will apply additional filter(-s) using Collection Interface which will filter out some data - we will break Layered Navigation. And if we could re-calculate getTotalCount, it's impossible to make that for LN. 
So, as for me it's okay to leave everything as is.